dev/partkey.md

```diff
@@ -0,0 +1,165 @@
+---
+numbersections: true
+title: "Algorand Key Specification"
+date: \today
+abstract: >
+    This document specifies the list of keys and their capabilities in Algorand.
+---
+
+Overview
+========
+
+An algorand node uses two kind of cryptographic keys:
+
+ - _participation keys_, a set of keys used for authentication, i.e. identify an 
+    account. Algorand uses a hierarchical (two\-level) identity encryption scheme that 
+    ensures [forward security](https://en.wikipedia.org/wiki/Forward_secrecy), 
+    which will be detailed in next section. 
+
+ - _selection credential_, key used for proving membership of selection. 
+
+A vote is valid only if both the participation keys and the selection
+credential are valid.
+
+Participation Keys 
+==================
+
+\newcommand \KeyDilution {\mathrm{KeyDilution}}
+\newcommand \Batch {\mathrm{Batch}}
+\newcommand \Offset {\mathrm{Offset}}
+
+Algorand's Two\-level Ephemeral Signature Scheme for Authentication
+------------------------------------------------------------------
+
+An ephemeral subkey is a key pair that produces one\-time signature 
+for messages. It must be deleted after use to ensure forward security.
+Algorand's ephemeral subkeys 
+uses [Ed25519 public\-key signature system](https://ed25519.cr.yp.to/).
+
+Algorand uses a two\-level ephemeral signature scheme.
+Instead of signing voting messages directly, an algorand account uses her
+master key to sign an intermediate ephemeral sub-key. 
+This intemediate ephemeral sub-key signs a batch of leaf level ephemeral 
+sub-keys. Hence, each intermediate ephemeral sub-key is associate with a
+batch number ($\Batch$), and each leaf ephemeral sub-key is associate with a
+batch number (of its parent key) and an offset ($\Offset$, denotes its offset
+in current batch). A voting message is signed hierarchically: 
+the master key $\rightarrow$ batch sub\-key $\rightarrow$ leaf sub-key
+$\rightarrow$ voting message (more details in next sub-section: One\-time 
+Signature).
+
+Each leaf level ephemeral sub-key is used only once, and will be deleted and 
```

Comment:
Not sure to fully understand what this sentence mean.

Reply:
I rephrased that.